pkg/controllers/coscheduling/podgroup_controller.go

```diff
@@ -47,16 +66,158 @@ type PodGroupReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *PodGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx)
+	pg := &schedv1alpha1.PodGroup{}
+	if err := r.Get(ctx, req.NamespacedName, pg); err != nil {
+		if apierrs.IsNotFound(err) {
+			log.V(5).Info("Pod group has been deleted")
+			return ctrl.Result{}, nil
+		}
+		log.V(3).Error(err, "Unable to retrieve pod group")
+		return ctrl.Result{}, err
+	}
+
+	if pg.Status.Phase == schedv1alpha1.PodGroupFinished ||
+		pg.Status.Phase == schedv1alpha1.PodGroupFailed {
+		return ctrl.Result{}, nil
+	}
+	// If startScheduleTime - createTime > 2days,
+	// do not reconcile again because pod may have been GCed
+	if pg.Status.Scheduled == pg.Spec.MinMember && pg.Status.Running == 0 &&
+		pg.Status.ScheduleStartTime.Sub(pg.CreationTimestamp.Time) > 48*time.Hour {
+		r.recorder.Event(pg, v1.EventTypeWarning,
+			"Timeout", "schedule time longer than 48 hours")
+		return ctrl.Result{}, nil
+	}
+
+	pgCopy := pg.DeepCopy()
+	podList := &corev1.PodList{}
+	if err := r.List(ctx, podList,
+		client.MatchingLabelsSelector{
+			Selector: labels.Set(map[string]string{
+				schedv1alpha1.PodGroupLabel: pgCopy.Name}).AsSelector(),
+		}); err != nil {
+		log.Error(err, "List pods for group failed")
```

Comment:
also need a print namespace and name of this PodGroup

Reply:
the logger from controller-runtime context is printing namespace and name by default

Reply:
OK